CC-21931 Integration guide for Warehouse User epic (#1512)
spryker/spryker-docs: _includes/pbc/all/install-features/202304.0/install-warehouse-user-management-feature.md

> This document describes how to integrate the Warehouse User Management feature into a Spryker project.

Reviewer: Why is this feature absent in the Release Groups app?

Author: It is a new feature, it will appear there after the release.

| StockCriteriaFilter.stockIds                    | property | created    | src/Generated/Shared/Transfer/StockCriteriaFilter                             |
| Collection                                      | class    | deprecated | src/Generated/Shared/Transfer/Collection                                      |
| UserCriteria.idUser                             | property | deprecated | src/Generated/Shared/Transfer/UserCriteria                                    |
| UserCriteria.email                              | property | deprecated | src/Generated/Shared/Transfer/UserCriteria                                    |
| UserCriteria.userReference                      | property | deprecated | src/Generated/Shared/Transfer/UserCriteria                                    |
| UserCriteria.withExpanders                      | property | deprecated | src/Generated/Shared/Transfer/UserCriteria                                    |
| StockCriteriaFilter.idStock                     | property | deprecated | src/Generated/Shared/Transfer/StockCriteriaFilter                             |

{% endinfo_block %}

### 3) Set up configuration

> If you want to make `warehouse-user-assignments` resource protected, adjust the protected paths configuration:

Reviewer: is this step optional?
If _yes_, then rewrite it as follows:
```suggestion
Optional: To make `warehouse-user-assignments` resource protected, adjust the protected paths configuration:
```
if _no_, then rewrite it as follows:
```suggestion
To make `warehouse-user-assignments` resource protected, adjust the protected paths configuration:
```

warehouse_user_assignment.validation.user_not_found,Benutzer nicht gefunden.,de_DE
warehouse_user_assignment.validation.warehouse_not_found,Warehouse not found.,en_US
warehouse_user_assignment.validation.warehouse_not_found,Lager nicht gefunden.,de_DE
warehouse_user_assignment.validation.warehouse_user_assignment_not_found,Warehouse user assignment not found.,en_US
warehouse_user_assignment.validation.warehouse_user_assignment_not_found,Lagerbenutzerzuweisung nicht gefunden.,de_DE
warehouse_user_assignment.validation.too_many_active_warehouse_assignments,User has too many active warehouse assignments.,en_US
warehouse_user_assignment.validation.too_many_active_warehouse_assignments,Dem Benutzer sind zu viele aktive Läger zugewiesen.,de_DE
warehouse_user_assignment.validation.warehouse_user_assignment_already_exists,Warehouse user assignment already exists.,en_US
warehouse_user_assignment.validation.warehouse_user_assignment_already_exists,Lagerbenutzerzuweisung existiert bereits.,de_DE
```

2. Import data:

```bash
console data:import glossary
```

3. Add Zed translations:

Generate a new translation cache for Zed:

```bash
console translator:generate-cache
```

> {% info_block warningBox "Verification" %}

Reviewer: I've rewritten the Verification steps, so review them and make sure they are correct since I don't have this feature to be able to test it in the Back Office on my own.

Author: Looks good to me 👍

    {
        return [
            new WarehouseUserLoginRestrictionPlugin(),
        ];
    }
}

```

**src/Pyz/Zed/User/UserDependencyProvider.php**

```php
<?php

namespace Pyz\Zed\User;

use Spryker\Zed\User\UserDependencyProvider as SprykerUserDependencyProvider;
use Spryker\Zed\WarehouseUserGui\Communication\Plugin\User\WarehouseUserAssignmentUserFormExpanderPlugin;
use Spryker\Zed\WarehouseUserGui\Communication\Plugin\User\WarehouseUserAssignmentUserTableActionExpanderPlugin;

class UserDependencyProvider extends SprykerUserDependencyProvider
{
    /**
     * @return array<\Spryker\Zed\UserExtension\Dependency\Plugin\UserTableActionExpanderPluginInterface>
     */
    protected function getUserTableActionExpanderPlugins(): array
    {
        return [
            new WarehouseUserAssignmentUserTableActionExpanderPlugin(),
        ];
    }

    /**
     * @return array<\Spryker\Zed\UserExtension\Dependency\Plugin\UserFormExpanderPluginInterface>
     */
    protected function getUserFormExpanderPlugins(): array
    {
        return [
            new WarehouseUserAssignmentUserFormExpanderPlugin(),
        ];
    }
}

```

> ---

Reviewer: ```suggestion
```

> {% info_block warningBox "Verification" %}

Reviewer: I've rewritten the Verification steps, so review them and make sure they are correct since I don't have this feature to be able to test it in the Back Office on my own.

Author: Also looks good 👍
